### PR TITLE
Dog Access Level Information

### PR DESCRIPTION
**Pull request by qad114**
## Dog Access Level Information

Issue Number(s): #123

Adjusts the information returned to the user depending on their access level and association with the dog(s). 

### Checklist

- [ ] Requirements have been implemented
- [ ] Acceptance criteria is met
- [ ] Database schema [docs](https://www.notion.so/gtbitsofgood/Database-Schema-Docs-fd377ded10b54890b65fd99025488cee?pvs=4) have been updated or are not necessary
- [ ] Relevant reviewers (EM) have been assigned to this PR

### How To Test

On the dashboard, ensure that only dogs associated with the user are displayed. Also ensure that, if a dog with a volunteer/partner association is clicked, then only limited information is displayed.

### CODE REVIEW

File: src/pages/api/dogs/[id].js [hunk 1]
```diff
@@ -51,6 +53,27 @@ export default async function handler(req, res) {
         return;
       }
 
+      const token = await getToken({ req });
```
Comment (afazio1) [inline src/pages/api/dogs/[id].js]: use `const session = await getServerSession(req, res, authOptions);` for consistency

File: src/pages/api/dogs/[id].js [hunk 2]
```diff
@@ -51,6 +53,27 @@ export default async function handler(req, res) {
...
+      const user = await getUserById(token.sub);
+      // Only filter keys if association is partner/volunteer and nothing else
+      // Throw an error if there is no association
+      if (
+        user.role !== "Admin" &&
```
Comment (afazio1) [inline src/pages/api/dogs/[id].js]: make sure `"Manager"`s can see all dogs as well

**Review by afazio1 — CHANGES_REQUESTED** [review]
Great work! See comments for a few minor changes

**Review by laurenmcp — COMMENTED** [review]
Nice job Hamza!! Looks great to me provided that alexa's requested changes are implemented.

**Review by afazio1 — APPROVED** [review]
Love it when it just works 👍 😎

**Pull request merged**